source/core/src/main/com/csse3200/game/components/quests/QuestManager.java

```diff
@@ -41,150 +39,257 @@ public class QuestManager extends Component {
 
     private final Entity player;
 
-    private final DialogueBoxService dialogueBoxService;
-
-
 
     /**Constructs questManager instance */
     public QuestManager(Entity player) {
-        this.quests = new HashMap<>();
+        this.quests = new LinkedHashMap<>();
         this.player = player;
         this.relevantQuests = Map.of(
                 "Cow", new String[]{"2 Task Quest"}
         );
-        this.dialogueBoxService = ServiceLocator.getDialogueBoxService();
+//        this.dialogueBoxService = ServiceLocator.getDialogueBoxService();
```

Comment:
maybe remove this line if it is not needed

Reply:
thanks. fixing now :)